.github/workflows/ci-deep.yml

```diff
@@ -32,188 +32,46 @@ on:
-    steps:
-      - name: "Check out the repo"
-        uses: "actions/checkout@v4"
-
-      - name: "Install Foundry"
-        uses: "foundry-rs/foundry-toolchain@v1"
-
-      - name: "Install Bun"
-        uses: "oven-sh/setup-bun@v1"
-
-      - name: "Install the Node.js dependencies"
-        run: "bun install --frozen-lockfile"
-
-      - name: "Lint the code"
-        run: "bun run lint"
-
-      - name: "Add lint summary"
-        run: |
-          echo "## Lint result" >> $GITHUB_STEP_SUMMARY
-          echo "✅ Passed" >> $GITHUB_STEP_SUMMARY
+    uses: "sablier-labs/reusable-workflows/.github/workflows/forge-lint.yml@main"
 
   build:
-    runs-on: "ubuntu-latest"
-    steps:
-      - name: "Check out the repo"
-        uses: "actions/checkout@v4"
-
-      - name: "Install Foundry"
-        uses: "foundry-rs/foundry-toolchain@v1"
-
-      - name: "Install Bun"
-        uses: "oven-sh/setup-bun@v1"
-
-      - name: "Install the Node.js dependencies"
-        run: "bun install --frozen-lockfile"
-
-      - name: "Show the Foundry config"
-        run: "forge config"
-
-      - name: "Produce an optimized build with --via-ir"
-        run: "FOUNDRY_PROFILE=optimized forge build"
-
-      - name: "Build the test contracts"
-        run: "FOUNDRY_PROFILE=test-optimized forge build"
-
-      - name: "Cache the build and the node modules so that they can be re-used by the other jobs"
-        uses: "actions/cache/save@v3"
-        with:
-          key: "build-and-modules-${{ github.sha }}"
-          path: |
-            cache
-            node_modules
-            out
-            out-optimized
-
-      - name: "Add build summary"
-        run: |
-          echo "## Build result" >> $GITHUB_STEP_SUMMARY
-          echo "✅ Passed" >> $GITHUB_STEP_SUMMARY
+    uses: "sablier-labs/reusable-workflows/.github/workflows/forge-build.yml@main"
 
   test-unit:
-    env:
-      FOUNDRY_FUZZ_RUNS: ${{ inputs.unitFuzzRuns || '100000' }}
     needs: ["lint", "build"]
-    runs-on: "ubuntu-latest"
-    steps:
-      - name: "Check out the repo"
-        uses: "actions/checkout@v4"
-
-      - name: "Install Foundry"
-        uses: "foundry-rs/foundry-toolchain@v1"
-
-      - name: "Restore the cached build and the node modules"
-        uses: "actions/cache/restore@v3"
-        with:
-          fail-on-cache-miss: true
-          key: "build-and-modules-${{ github.sha }}"
-          path: |
-            cache
-            node_modules
-            out
-            out-optimized
-
-      - name: "Run the unit tests against the optimized build"
-        run: "FOUNDRY_PROFILE=test-optimized forge test --match-path \"test/unit\""
-
-      - name: "Add test summary"
-        run: |
-          echo "## Unit tests result" >> $GITHUB_STEP_SUMMARY
-          echo "✅ Passed" >> $GITHUB_STEP_SUMMARY
+    uses: "sablier-labs/reusable-workflows/.github/workflows/forge-test.yml@main"
+    with:
+      foundry-fuzz-runs: ${{ inputs.unitFuzzRuns || 100000 }}
```

Comment:
Same comment as [this](https://github.com/sablier-labs/v2-periphery/pull/259#discussion_r1450186241).

Reply:
Answered there:

https://github.com/sablier-labs/v2-periphery/pull/259#discussion_r1450263171